src/wp-includes/option.php

```diff
@@ -1106,6 +1113,71 @@ function delete_option( $option ) {
 	return false;
 }
 
+/**
+ * Determine the appropriate autoload value based on input.
+ *
+ * This function checks the provided autoload value and returns a standardized
+ * value ('yes', 'no', 'default-yes', or 'default-no') based on specific conditions.
+ *
+ * @since 6.4.0
+ *
+ * @param mixed $autoload The autoload value to evaluate.
+ *
+ * @return string The standardized autoload value: 'yes', 'no', 'default-yes', or 'default-no'.
+ */
+function sanitize_autoload( $autoload ) {
+	// Check if autoload is a boolean.
+	if ( is_bool( $autoload ) ) {
+		return $autoload ? 'yes' : 'no';
+	}
+
+	switch ( $autoload ) {
+		case 'no':
+		case 'yes':
+		case 'default-yes':
+		case 'default-no':
+			return $autoload;
+		default:
+			return 'default-yes';
+	}
+}
+
+/**
+ * Check the size of an option value and compare it to the maximum allowed size.
+ *
+ * This function serializes the given value and checks its size against the maximum allowed
+ * option size defined by the 'max_option_size' filter.
+ *
+ * @since 6.4.0
+ *
+ * @param string $name  The name of the option.
+ * @param mixed  $value The value of the option to be checked.
+ *
+ * @return string Returns 'default-no' if the size exceeds the maximum allowed size, or 'default-yes' otherwise.
+ */
+function sanitize_option_size($name, $value ) {
+	// Serialize the value and check its size against the maximum allowed option size.
+	$serialized_value = maybe_serialize( $value );
+	$size             = strlen( $serialized_value );
+
+	/**
+	 * Filters the maximum size of option value in bytes.
+	 *
+	 * @since 6.4.0
+	 *
+	 * @param int    $max_option_size The option-size threshold, in bytes. Default 150000.
+	 * @param string $name            The name of the option.
+	 * @param mixed  $value           The value of the option.
+	 */
+	$max_option_size = (int) apply_filters( 'max_option_size', 150000, $name, $value );
```

Comment:
There's a WPCS violation here, also let's use `$option` for the parameter name as it is used elsewhere.
```suggestion
 * @param string $option The name of the option.
 * @param mixed  $value  The value of the option to be checked.
 * @return string Returns 'default-no' if the size exceeds the maximum allowed size, or 'default-yes' otherwise.
 */
function sanitize_option_size( $option, $value ) {
	// Serialize the value and check its size against the maximum allowed option size.
	$serialized_value = maybe_serialize( $value );
	$size             = strlen( $serialized_value );

	/**
	 * Filters the maximum size of option value in bytes.
	 *
	 * @since 6.4.0
	 *
	 * @param int    $max_option_size The option-size threshold, in bytes. Default 150000.
	 * @param string $option          The name of the option.
	 * @param mixed  $value           The value of the option.
	 */
	$max_option_size = (int) apply_filters( 'max_option_size', 150000, $option, $value );
```